indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/ScalingStats.java

```diff
@@ -65,6 +65,15 @@ public ScalingStats(int capacity)
     }
   }
 
+  public ScalingStats(ScalingStats stats1, ScalingStats stats2)
+  {
+    this(stats1.recentEvents.size() + stats2.recentEvents.size());
+    synchronized (lock) {
```

Comment:
It can matter if the object is being accessed by multiple threads. Only final fields are guaranteed memory consistent after the constructor. Any objects in a list in the final field (for example) are not guaranteed memory consistent.